application-licensing-licensor/application-licensing-licensor-api/src/main/java/com/xwiki/licensing/internal/TrialLicenseGenerator.java

```diff
@@ -0,0 +1,251 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package com.xwiki.licensing.internal;
+
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.util.List;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Provider;
+import javax.inject.Singleton;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.lang3.exception.ExceptionUtils;
+import org.apache.http.client.utils.URIBuilder;
+import org.slf4j.Logger;
+import org.xwiki.component.annotation.Component;
+import org.xwiki.crypto.BinaryStringEncoder;
+import org.xwiki.extension.ExtensionId;
+import org.xwiki.instance.InstanceIdManager;
+import org.xwiki.properties.converter.Converter;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.xpn.xwiki.XWikiContext;
+import com.xwiki.licensing.License;
+import com.xwiki.licensing.LicenseManager;
+import com.xwiki.licensing.LicensedExtensionManager;
+import com.xwiki.licensing.LicensingConfiguration;
+import com.xwiki.licensing.Licensor;
+
+/**
+ * Helper methods for generating a trial license and updating it.
+ *
+ * @since 1.17
+ * @version $Id$
+ */
+@Component(roles = TrialLicenseGenerator.class)
+@Singleton
+public class TrialLicenseGenerator
+{
+    private static final String FAILED_TO_ADD_TRIAL_LICENSE = "Failed to add trial license";
+
+    private static final String FEATURE_ID = "featureId";
+
+    private static final String INSTANCE_ID = "instanceId";
+
+    @Inject
+    private Logger logger;
+
+    @Inject
+    private Provider<InstanceIdManager> instanceIdManagerProvider;
+
+    @Inject
+    private UserCounter userCounter;
+
+    @Inject
+    private LicensedExtensionManager licensedExtensionManager;
+
+    @Inject
+    private LicensingConfiguration licensingConfig;
+
+    @Inject
+    private Provider<Licensor> licensorProvider;
+
+    @Inject
+    private Provider<LicenseManager> licenseManagerProvider;
+
+    @Inject
+    @Named("Base64")
+    private BinaryStringEncoder base64decoder;
+
+    @Inject
+    private Converter<License> converter;
+
+    @Inject
+    private Provider<XWikiContext> contextProvider;
+
+    /**
+     * Generate trial license for the given extension.
+     *
+     * @param extensionId the extension for which to generate a trial license
+     */
+    public void generateTrialLicense(ExtensionId extensionId)
+    {
+        try {
+            URL trialURL = getTrialURL(extensionId);
+            if (trialURL == null) {
+                logger.info(FAILED_TO_ADD_TRIAL_LICENSE);
```

Comment:
I would use debug level here and give move details: "Failed to add trial license because the licensor configuration is not complete. Check your store trial URL and owner details."